### PR TITLE
:zap: feat(v0.1.0): new design and allow updating custom IP

### PR DESCRIPTION
**Pull request by aashutoshrathi**
| Design |  Actual  |
|---|---|
| image  | image |

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ This is a simple extension to upsert your IP addresses into a [MongoDB NAL](http
 
 ## Motivation 🤔
 
-Unstable internet connection is pain and when you have to access some database from local, but your IP changes very often, it becomes a pain to update the NAL to whitelist your IP.
+Unstable internet connection is pain and when you have to access some database from local, but your IP changes very often, it becomes a pain to update the NAL to allow your IP.
```
Comment (mohitkyadav) [inline README.md]:
```suggestion
Unstable internet connection is already a pain and when you have to access some database from local, but your IP changes very often, it becomes a pain in the ass to update the NAL to allow your IP.
```

File: manifest.json [hunk 2]
```diff
@@ -1,9 +1,9 @@
 {
   "name": "Mongo NAL Updater",
   "author": "Aashutosh Rathi",
-  "version": "0.0.1",
+  "version": "0.1.0",
```
Comment (mohitkyadav) [inline manifest.json:4]: big bump

File: manifest.json [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
   "manifest_version": 3,
-  "description": "Adds or update Mongo NAL Whitelist entry",
+  "description": "Adds or update Mongo Network access allowed list entry",
```
Comment (mohitkyadav) [inline manifest.json]:
```suggestion
  "description": "Adds or updates Mongo Network access allowed list entry",
```

File: src/content.js [hunk 4]
```diff
@@ -2,14 +2,22 @@ const sleep = (milliseconds) => {
...
 };
 
-const clickOnAddCurrentIP = async () => {
-  const addCurrentButton = document.querySelector(
-    "button[name='addCurrentIpAddress']"
-  );
-  if (addCurrentButton) {
-    addCurrentButton.click();
-    await sleep(1000);
+const clickOnAddCurrentIP = async (ip) => {
```
Comment (mohitkyadav) [inline src/content.js]:
```suggestion
const handleAddCurrentIPClick = async (ip) => {
```

File: src/content.js [hunk 5]
```diff
@@ -2,14 +2,22 @@ const sleep = (milliseconds) => {
...
+      "button[name='addCurrentIpAddress']"
+    );
+    if (addCurrentButton) {
+      addCurrentButton.click();
+      await sleep(1000);
+    }
+    return;
   }
+  const ipField = document.querySelector('[name="networkPermissionEntry"]');
+  if (ipField) {
```
Comment (mohitkyadav) [inline src/content.js:19]:
```suggestion
  
  if (ipField) {
```

File: src/content.js [hunk 6]
```diff
@@ -2,14 +2,22 @@ const sleep = (milliseconds) => {
...
-  );
-  if (addCurrentButton) {
-    addCurrentButton.click();
-    await sleep(1000);
+const clickOnAddCurrentIP = async (ip) => {
+  if (!ip) {
+    const addCurrentButton = document.querySelector(
+      "button[name='addCurrentIpAddress']"
+    );
+    if (addCurrentButton) {
```
Comment (mohitkyadav) [inline src/content.js:11]:
```suggestion

    if (addCurrentButton) {
```

File: src/content.js [hunk 7]
```diff
@@ -2,14 +2,22 @@ const sleep = (milliseconds) => {
...
+    return;
   }
+  const ipField = document.querySelector('[name="networkPermissionEntry"]');
+  if (ipField) {
+    ipField.value = ip;
+  }
+
 };
 
 const clickOnSaveButton = async () => {
```
Comment (mohitkyadav) [inline src/content.js]:
```suggestion
const handleSaveButtonClick = async () => {
```

File: src/options/index.html [hunk 8]
```diff
@@ -1,12 +1,45 @@
...
+    <div class="hflex align-center">
+      <div class="icon-container">
+        <img class="icon" src="../../icons/icon.png" />
+      </div>
+      <div class="vflex text-center">
+        <p class="heading">MONGO IP UPDATER</p>
+        <!-- <i class="lazy">&nbsp;&nbsp;&nbsp;for the lazy</i> -->
+      </div>
+    </div>
+    <div>
```
Comment (mohitkyadav) [inline src/options/index.html:18]: Give some spacing here, 12px/16px.

**Review by mohitkyadav — COMMENTED** [review]
Make the button take full width. And IP input looks ugly.

File: manifest.json (hunk 2 above)
Comment (aashutoshrathi) [reply manifest.json:4]: Only aim for the moon

File: README.md (hunk 1 above)
Comment (aashutoshrathi) [reply README.md]: hahah! shouldn't we say `arse`?

File: manifest.json (hunk 3 above)
Comment (aashutoshrathi) [reply manifest.json]: the-simpsons-gif

File: src/content.js (hunk 4 above)
Comment (aashutoshrathi) [reply src/content.js]: nah nah! it's not handling, it's actually clicking 🤣

**Comment by aashutoshrathi** [comment]
> Make the button take full width. And IP input looks ugly.

Senpai is rude! Let me try fixing that

File: src/content.js (hunk 7 above)
Comment (aashutoshrathi) [reply src/content.js]: this is also action

File: src/options/index.html (hunk 8 above)
Comment (aashutoshrathi) [reply src/options/index.html:18]: where?

**Comment by aashutoshrathi** [comment]
**PS:** Dropping all words like `whitelist` for obvious reasons

**Comment by aashutoshrathi** [comment]
We are ready for review Round 2 @mohitkyadav 
steins-gate-itaru-gif

**Review by mohitkyadav — APPROVED** [review]

**Pull request merged**